qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1180,29 +1318,17 @@ def copy(self, copy_results: bool = True) -> "DbExperimentDataV1":
         # Copy circuit result data and jobs
         with self._data.lock:  # Hold the lock so no new data can be added.
             new_instance._data = self._data.copy_object()
-            for orig_kwargs, fut in self._job_futures.copy():
-                if fut.done():
-                    continue
-                # We cannot deep copy orig_kwargs because it contains a Job which
-                # inherits an abstract class.
-                extra_kwargs = {}
-                for key, val in orig_kwargs.items():
-                    if key not in ["jobs", "timeout"]:
-                        extra_kwargs[key] = val
-
-                new_instance.add_data(
-                    data=orig_kwargs["jobs"],
-                    timeout=orig_kwargs["timeout"],
-                    **extra_kwargs,
-                )
+            for jid, fut in self._job_futures.items():
```

Comment:
This doesn't cause edge case? I guess job is running on separate thread so even if we lock the container this could change status (e.g. nanosec type bug). I don't completely understand relationship of these processes but l1323 is branched by `fut` which is likely locked in the context so should be safe? Or still there is possibility that add data method is called in source object after copy_object is called?